src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs

```diff
@@ -29,15 +30,23 @@ class ManagedResourceParser : ResourceParser
 
 		public string ResourceFlagFile { get; set; }
 
-		void SortMembers (CodeTypeDeclaration decl)
+		void SortMembers (CodeTypeDeclaration decl, StringComparison stringComparison = StringComparison.OrdinalIgnoreCase)
 		{
 			CodeTypeMember [] members = new CodeTypeMember [decl.Members.Count];
 			decl.Members.CopyTo (members, 0);
 			decl.Members.Clear ();
-			Array.Sort (members, (x, y) => string.Compare (x.Name, y.Name, StringComparison.OrdinalIgnoreCase));
+			Array.Sort (members, (x, y) => string.Compare (x.Name, y.Name, stringComparison));
 			decl.Members.AddRange (members);
 		}
 
+		IEnumerable<CodeTypeMember> SortedMembers (CodeTypeDeclaration decl, StringComparison stringComparison = StringComparison.Ordinal)
+		{
+			CodeTypeMember [] members = new CodeTypeMember [decl.Members.Count];
+			decl.Members.CopyTo (members, 0);
+			Array.Sort (members, (x, y) => string.Compare (x.Name, y.Name, stringComparison));
+			return members;
+		}
```

Comment:
Would a regular System.Linq `OrderBy` be just as good here? It probably does the exact same thing, just less code. It might also not allocate the entire array and uses `yield return`.